common/flatpak-dir.c

```diff
@@ -8886,6 +8914,7 @@ flatpak_dir_update_remote_configuration (FlatpakDir   *self,
     return flatpak_dir_update_remote_configuration_for_repo_metadata (self, remote, summary, FALSE, NULL, cancellable, error);
 }
 
+/* TODO: Move these sizes out to per-ref or per-commit metadata */
```

Comment:
This comment scares me a bit. We don't want to be in the situation where we have gnome-software blocking on startup until it has downloaded all the commit objects in all remotes before it starts showing anything, that will be very slow.

We already store this info in the commit metadata, but we propagate it to the summary for this reason. The summary file is a single, reasonlably compact file that we can download once and the have all info we need.

This, in combination with using unsigned summaries implies that we have to keep track of what the summary metadata said about a specific revision, and then verify that it was actually true when we pull the commit. I.e. we should not only verify that the signed commit has the same ref and collection id, but also that the metadata is identical to what was in the summary.